x-pack/legacy/plugins/ml/server/new_platform/job_validation_schema.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { anomalyDetectionJobSchema } from './anomaly_detectors_schema';
+
+export const estimateBucketSpanSchema = {
+  aggTypes: schema.arrayOf(schema.nullable(schema.string())),
```

Comment:
Question: don't know the functional domain, but `schema.arrayOf(schema.nullable(schema.string()))` looks strange to me. can/should it really accept something like `['foo', null, 'bar', null]` ?

Reply:
Yep - so this was based on the `BucketEstimatorData` type interface (https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/ml/public/application/services/ml_api_service/index.d.ts#L48) used to define the type of data sent to the bucket span estimator endpoint.

https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/bucket_span_estimator/estimate_bucket_span.ts#L31

https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/bucket_span_estimator/estimate_bucket_span.ts#L68